go-controller/pkg/ovn/egressip.go

```diff
@@ -1614,16 +1622,23 @@ func (e *egressIPController) deleteEgressReroutePolicy(filterOption, egressIPNam
 		{
 			Model: &logicalRouterPolicy,
 			ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool {
-				return lrp.Match == filterOption && lrp.Priority == types.EgressIPReroutePriority && lrp.ExternalIDs["name"] == egressIPName
+				if lrp.Match != filterOption || lrp.Priority != types.EgressIPReroutePriority || lrp.ExternalIDs["name"] != egressIPName {
+					return false
+				}
+				// Ensure that gatewayRouterIP is part of nexthops
+				for _, nextHop := range lrp.Nexthops {
+					if nextHop == gatewayRouterIP {
+						return true
+					}
+				}
+				return false
```

Comment:
I don't think the current code is wrong here and needs a change, actually. I think what you need to be solving is sync function, which is what the BZ was about right?

Reply:
@alexanderConstantinescu : This specific logic is about removing the gatewayRouterIP from the policy. Removal of the entire policy is done when we remove the reference to the lrp from the router. That is done in the code below.
The core of the removal is done by the mutate delete operation on the nextops:

```
logicalRouterPolicy := nbdb.LogicalRouterPolicy{
		Nexthops: []string{gatewayRouterIP},
}
```

Since it is okay remove an entry that does not exist in the list, you are right that we could simply have it removed and not bother checking if it is there. Is that what you propose we should do? I can do that. Less is more. :)

Reply:
Done. Let me know if that is not what you intended to have, please.